transport-native-io_uring/src/main/java/io/netty/channel/uring/IOUringCompletionQueue.java

```diff
@@ -0,0 +1,137 @@
+/*
+ * Copyright 2020 The Netty Project
+ *
+ * The Netty Project licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+package io.netty.channel.uring;
+
+import io.netty.util.internal.PlatformDependent;
+
+final class IOUringCompletionQueue {
+
+  //these offsets are used to access specific properties
+  //CQE (https://github.com/axboe/liburing/blob/master/src/include/liburing/io_uring.h#L162)
+  private static final int CQE_USER_DATA_FIELD = 0;
+  private static final int CQE_RES_FIELD = 8;
+  private static final int CQE_FLAGS_FIELD = 12;
+
+  private static final int CQE_SIZE = 16;
+
+  private static final int IORING_ENTER_GETEVENTS = 1;
+
+  //these unsigned integer pointers(shared with the kernel) will be changed by the kernel
+  private final long kHeadAddress;
+  private final long kTailAddress;
+  private final long kringMaskAddress;
+  private final long kringEntries;
+  private final long kOverflowAddress;
+
+  private final long completionQueueArrayAddress;
+
+  private final int ringSize;
+  private final long ringAddress;
+  private final int ringFd;
+
+  IOUringCompletionQueue(long kHeadAddress, long kTailAddress, long kringMaskAddress, long kringEntries,
+      long kOverflowAddress, long completionQueueArrayAddress, int ringSize, long ringAddress, int ringFd) {
+    this.kHeadAddress = kHeadAddress;
+    this.kTailAddress = kTailAddress;
+    this.kringMaskAddress = kringMaskAddress;
+    this.kringEntries = kringEntries;
+    this.kOverflowAddress = kOverflowAddress;
+    this.completionQueueArrayAddress = completionQueueArrayAddress;
+    this.ringSize = ringSize;
+    this.ringAddress = ringAddress;
+    this.ringFd = ringFd;
+  }
+
+  public IOUringCqe peek() {
+    long cqe = 0;
```

Comment:
nit: this definition is not needed and you can define it directly on assignment in the if block below.